src/integration-tests/test_admin_res_log.py

```diff
@@ -0,0 +1,181 @@
+This suite of test cases verifies that only the node that the admin
+connects to send the response back to the admin, and the corresponding
+repsonse is logged.
+"""
+
+import blazingmq.dev.it.testconstants as tc
+from blazingmq.dev.it.fixtures import (
+    multi_node,
+    Cluster,
+)
+from blazingmq.dev.it.process.admin import AdminClient
+from blazingmq.dev.it.process.client import Client
+
+TIMEOUT = 2
+
+
+def test_adminsession_res_log_stat(multi_node: Cluster):
+    """
+    Test: STAT SHOW command: response message only generate on the node itself
+    """
+
+    admin = AdminClient()
+
+    # find the first two nodes which are not the known leader
+    member1 = next(
+        node for node in multi_node.nodes() if node != node.last_known_leader
+    )
+    member2 = next(
+        node
+        for node in multi_node.nodes()
+        if node != member1 and node != node.last_known_leader
+    )
+    leader = member1.last_known_leader
+
+    # open a domain
+    proxies = multi_node.proxy_cycle()
+    proxy = next(proxies)
+    producer: Client = proxy.create_client("producer")
+    producer.open(tc.URI_FANOUT, flags=["write,ack"], succeed=True)
+
+    # connect and send request to primary
+    admin.connect(leader.config.host, int(leader.config.port))
+    res = admin.send_admin(f"STAT SHOW")
+
+    assert ":::::::::: :::::::::: DOMAINQUEUES >>" in res
+
+    assert leader.capture("Send response message", TIMEOUT)
+    assert not member1.capture("Send response message", TIMEOUT)
+    assert not member2.capture("Send response message", TIMEOUT)
+
+    admin.stop()
+
+    # connect and send request to member1
+    admin.connect(member1.config.host, int(member1.config.port))
+    res = admin.send_admin(f"STAT SHOW")
+
+    assert ":::::::::: :::::::::: DOMAINQUEUES >>" in res
+
+    assert member1.capture("Send response message", TIMEOUT)
+    assert not leader.capture("Send response message", TIMEOUT)
+    assert not member2.capture("Send response message", TIMEOUT)
+
+    admin.stop()
+
+
+def test_adminsession_res_log_purge(multi_node: Cluster):
+    """
+    Test: PURGE command: response message only generate on the node itself
+    """
+
+    admin = AdminClient()
+
+    # find the first two nodes which are not the known leader
+    member1 = next(
+        node for node in multi_node.nodes() if node != node.last_known_leader
+    )
+    member2 = next(
+        node
+        for node in multi_node.nodes()
+        if node != member1 and node != node.last_known_leader
+    )
+    leader = member1.last_known_leader
+
+    # open a domain
+    proxies = multi_node.proxy_cycle()
+    proxy = next(proxies)
+    producer: Client = proxy.create_client("producer")
+    producer.open(tc.URI_FANOUT, flags=["write,ack"], succeed=True)
+
+    # connect and send request to primary
+    admin.connect(leader.config.host, int(leader.config.port))
+    res = admin.send_admin(f"DOMAINS DOMAIN {tc.DOMAIN_FANOUT} PURGE")
+
+    assert "Purged" in res
```

Comment:
Okay, I see.  If I run the test without a client connected, the response is "no queue purged", which doesn't match this assert.  It's fine with the the client connected then.